src/_components/AddAsset/AddAssetContainer.jsx

```diff
@@ -0,0 +1,173 @@
+import { createAsset } from '../../_actions/asset.actions';
+
+import {
+  filterSubCategories,
+  filterAssetTypes,
+  filterAssetMakes,
+  filterModelNumbers
+} from '../../_utils/filterDropdowns';
+
+class AddAssetContainer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      filteredSubCategories: [],
+      filteredAssetTypes: [],
+      filteredAssetMakes: [],
+      filteredModelNumbers: [],
+      modelNumber: 0,
+      serialNumber: '',
+      assetTag: '',
+      assets: typeof(props.assets) !== "undefined" ? props.assets: []
+    };
+  }
+
+  componentDidMount() {
+    if (_.isEmpty(this.props.categories)) {
+      this.props.loadCategories();
+    }
+    if (_.isEmpty(this.props.subcategories)) {
+      this.props.loadSubCategories();
+    }
+    if (_.isEmpty(this.props.assetTypes)) {
+      this.props.loadAssetTypes();
+    }
+    if (_.isEmpty(this.props.assetMakes)) {
+      this.props.loadAssetMakes();
+    }
+    if (_.isEmpty(this.props.modelNumbers)) {
+      this.props.loadModelNumbers();
+    }
+  }
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    if (nextProps.assets.length !== prevState.assets.length) {
+      ToastMessage.success({ message: 'Asset Saved Successfully' });
+      return {
+        filteredSubCategories: [],
+        filteredAssetTypes: [],
+        filteredAssetMakes: [],
+        filteredModelNumbers: [],
+        modelNumber: 0,
+        assets: nextProps.assets
+      }
+    }
+    return null;
+  }
+
+  handleDropdownChanges = (event, data) => {
+    const { name, value } = data;
+    const { subcategories, assetTypes, assetMakes, modelNumbers } = this.props;
+
+    if (name === 'asset-category') {
+      this.setState({
+        filteredSubCategories: filterSubCategories(subcategories, value)
+      });
+    } else if (name === 'asset-subcategory') {
+      this.setState({
+        filteredAssetTypes: filterAssetTypes(assetTypes, value)
+      });
+    } else if (name === 'asset-types') {
+      this.setState({
+        filteredAssetMakes: filterAssetMakes(assetMakes, value)
+      });
+    } else if (name === 'asset-makes') {
+      this.setState({
+        filteredModelNumbers: filterModelNumbers(modelNumbers, value)
+      });
+    }
+  };
+
+  onSelectModelNumber = (event, data) => {
+    this.setState({ modelNumber: data.value });
+  };
+
+  onAddSerialNumber = (event) => {
+    this.setState({ serialNumber: event.target.value });
+  };
+
+  onAddAssetTag = (event) => {
+    this.setState({ assetTag: event.target.value });
+  };
+
+  onCreateAsset = (event, data) => {
+    this.props.createAsset({
+      "asset_code": this.state.assetTag,
+      "serial_number": this.state.serialNumber,
+      "model_number": this.state.modelNumber
+    });
+    event.target.reset();
+  };
+
+  render() {
+    return (
+      <AddAssetComponent {...this.props}
+                         handleDropdownChanges={this.handleDropdownChanges}
```

Comment:
Is it your editor that added so many spaces for the props? It looked good the first time I reviewed it

Reply:
I think @Arthur236 and I are using 2 different IDE's. There must have been automatic readjustment with my editor. I have fixed it.